Tests/SentryTests/SentrySessionTests.m

```diff
@@ -0,0 +1,128 @@
+#import <XCTest/XCTest.h>
+#import "SentrySession.h"
+
+@interface SentrySessionTests : XCTestCase
+
+@end
+
+@implementation SentrySessionTests
+
+- (void)testInitDefaultValues {
+    SentrySession *session = [[SentrySession alloc] init];
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertEqual(1, session.sequence);
+    XCTAssertEqual(0, session.errors);
+    XCTAssertTrue(session.init);
+    XCTAssertNotNil(session.started);
+    XCTAssertEqual(kSentrySessionStatusOk, session.status);
+    XCTAssertNotNil(session.distinctId);
+
+    XCTAssertNil(session.timestamp);
+    XCTAssertNil(session.releaseName);
+    XCTAssertNil(session.environment);
+    XCTAssertNil(session.duration);
+}
+
+- (void)testSerializeDefaultValues {
+    SentrySession *expected = [[SentrySession alloc] init];
+    NSDictionary<NSString *, id>  *json = [expected serialize];
+    SentrySession *actual = [[SentrySession alloc] initWithJSONObject:json];
+
+    XCTAssertTrue([expected.sessionId isEqual:actual.sessionId]);
+    XCTAssertEqual(expected.sequence, actual.sequence);
+    XCTAssertEqual(expected.errors, actual.errors);
+    // TODO: get XCT happy XCTAssertEqual(expected.init, actual.init);
+    XCTAssertEqualWithAccuracy([expected.started timeIntervalSinceReferenceDate], [actual.started timeIntervalSinceReferenceDate], 1);
+    XCTAssertEqual(expected.status, actual.status);
+    XCTAssertEqual(expected.distinctId, actual.distinctId);
+    XCTAssertNil(expected.timestamp);
+    // Serialize session always have a timestamp (time of serialization)
+    XCTAssertNotNil(actual.timestamp);
+    XCTAssertNil(expected.releaseName);
+    XCTAssertNil(actual.releaseName);
+    XCTAssertNil(expected.environment);
+    XCTAssertNil(actual.environment);
+    XCTAssertNil(expected.duration);
+    XCTAssertNil(actual.duration);
+}
+
+- (void)testSerializeExtraFieldsEndedSessionWithNilStatus {
+    SentrySession *expected = [[SentrySession alloc] init];
+    NSDate *timestamp = [NSDate date];
```

Comment:
I usually prefer deterministic tests. This call here makes the test non-deterministic. Each time you run it the date is different. To change this we would need some kind of `DateProvider` that replaces those calls to `[NSDate date]`. In release it also calls `[NSDate date]`, but you could set it to any date when testing. What do you think?

Reply:
I agree, this approach was taken in other SDKs too. It would though need to be internal since the public API shouldn't be leaking these abstractions for the sake of testability.
Time constraints is the excuse for the pragmatism/hacks I'm throwing at this. Needs to be done by Friday.